File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    int letters = 0;
    int words = 0;
    int sentences = 0;

    string text = get_string("Text: ");

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else (text[i] == '.' || text[i] == '!' || text[i] == '?');
        {
            sentences++;
        }
    }
    int L = letters / words * 100;
    int S =  sentences / words * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1/n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+");
    }
    else
    {
        printf("Grade %i", index);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

letters = 0
words = 1
sentences = 0

text = get_string("Text: ")

# checks if symbol is letter, also checks for spaces(words) and finally for sentences, if yes, adds to counter


for i in range(len(text)):
    if text[i].isalpha():
        letters += 1

    elif text[i] == " ":
        words += 1

    elif text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1

# applying the formulas


L = letters / words * 100
S = sentences / words * 100

index = round(0.0588 * L - 0.296 * S - 15.8)


